service/src/event/migration.rs

```diff
@@ -295,33 +303,42 @@ impl EventBuilder {
     }
 
     fn add_root(&mut self, cid: Cid, data: Vec<u8>) {
-        self._add_block(cid, data, true)
+        self.blocks.push((cid, data, true));
     }
     fn add_block(&mut self, cid: Cid, data: Vec<u8>) {
-        self._add_block(cid, data, false)
-    }
-    fn _add_block(&mut self, cid: Cid, data: Vec<u8>, root: bool) {
-        self.blocks.push(EventBlockRaw {
-            event_cid: self.event_cid.to_bytes(),
-            codec: cid.codec() as i64,
-            root,
-            idx: self.blocks.len() as i32,
-            multihash: BlockHash::new(cid.hash().to_owned()),
-            bytes: data,
-        })
+        self.blocks.push((cid, data, false));
     }
 
-    async fn build(self, network: Network) -> Result<EventInsertable> {
+    async fn build(self, network: Network) -> Result<(EventId, Vec<u8>)> {
         let event_id = EventId::builder()
             .with_network(&network)
             .with_sep("model", &self.sep)
             .with_controller(&self.controller)
             .with_init(&self.init)
             .with_event(&self.event_cid)
             .build();
-        Ok(EventInsertable::try_new(
-            event_id,
-            EventInsertableBody::new(self.event_cid, self.blocks, false),
-        )?)
+
+        let body = self.build_car().await?;
+        Ok((event_id, body))
+    }
+
+    async fn build_car(&self) -> Result<Vec<u8>> {
+        let size = self
+            .blocks
+            .iter()
+            .fold(0, |sum, (_, row, _)| sum + row.len());
+        // Reconstruct the car file
+        let mut car = Vec::with_capacity(size + 100 * self.blocks.len());
+        let roots = self
+            .blocks
+            .iter()
+            .flat_map(|(cid, _, root)| if *root { Some(*cid) } else { None })
+            .collect::<Vec<_>>();
+        let mut writer = iroh_car::CarWriter::new(iroh_car::CarHeader::V1(roots.into()), &mut car);
```

Comment:
This isn't correct, not all blocks should be roots. Why not leverage the unvalidated::Event::encode_car here? We could make this EventBuilder a light wrapper on the unvalidated::Builder? That way we can keep the car file encoding in one place where it is well tested.

Reply:
Yeah, I like that idea. I kind of went qnd and will change it. This should only include roots (the third field in the tuple is whether it was a root and does encode as the tests expect) but it'd be a lot better to not include here at all.